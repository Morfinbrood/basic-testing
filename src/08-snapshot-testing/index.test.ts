import { generateLinkedList } from './index';

describe('generateLinkedList', () => {
  test('should generate linked list from values 1', () => {
    const values = [11, 22, 33];
    const expectedLinkedList = {
      value: values[0],
      next: {
        value: values[1],
        next: {
          value: values[2],
          next: {
            value: null,
            next: null,
          },
        },
      },
    };

    expect(generateLinkedList(values)).toStrictEqual(expectedLinkedList);
  });

  test('should generate linked list from values 2', () => {
    const values = ['el1', 'el2', 'el3'];

    const result = generateLinkedList(values);

    expect(result).toMatchSnapshot();
  });
});
