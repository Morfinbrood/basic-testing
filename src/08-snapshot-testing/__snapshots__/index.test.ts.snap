// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateLinkedList should generate linked list from values 2 1`] = `
{
  "next": {
    "next": {
      "next": {
        "next": null,
        "value": null,
      },
      "value": "el3",
    },
    "value": "el2",
  },
  "value": "el1",
}
`;
